spec/integration/session_spec.rb

```diff
@@ -956,6 +956,26 @@
         }.not_to raise_error
         expect(@session).to have_xpath("//a[@id='open-twice' and @confirmed='false']")
       end
+
+      it 'works with second window' do
+        @session.visit '/'
+
+        popup = @session.window_opened_by do
+          @session.execute_script <<-JS
+            window.open('/poltergeist/with_js', 'popup')
+          JS
+        end
+
+        @session.within_window(popup) do
+          expect {
+            @session.accept_confirm do
+              @session.click_link('Open for match')
+            end
+            expect(@session).to have_xpath("//a[@id='open-match' and @confirmed='true']")
+          }.not_to raise_error
+          @session.execute_script('window.close()')
```

Comment:
`popup.close()` outside the within_window block would be better

Reply:
fixed